src/test/java/com/github/fred84/queue/RedisDelayedEventServiceTest.java

```diff
@@ -245,22 +246,28 @@ void backPressureIsApplied() throws InterruptedException {
 
     @Test
     void verifyLogContext() throws InterruptedException {
-        var value = randomUUID().toString();
+        var collector = new ConcurrentHashMap<String, String>();
 
         eventService.addBlockingHandler(
                 DummyEvent.class,
-                e -> value.equals(MDC.get("key")),
+                e -> {
+                    collector.put(e.getId(), MDC.get("key"));
+                    return true;
+                },
                 1
         );
 
-        MDC.put("key", value);
-        eventService.enqueueWithDelayInner(new DummyEvent("1"), Duration.ZERO).block();
+        enqueue(3, id -> {
+            var str = Integer.toString(id);
```

Comment:
make event id more desriptive, e.g "event id:1" instead of just "1",
make context value more descriptive, e.g. "context value for key: 'key'",
add test for case when two different handlers use same (use different) context keys